lib/Preview.js

```diff
@@ -0,0 +1,181 @@
+  styles: compileTemplate('styles.css'),
+  embedJs: compileTemplate('embed.js'),
+};
+
+/**
+ * Reads all html files in a folder and generates the embeds.
+ */
+function generateEmbeds(config) {
+  glob(config.src + '/**/*.html', {}, (err, files) => {
+    files.forEach(file => generateEmbed(config, file));
+  });
+}
+
+/**
+ * Generates embeds for a given file.
+ */
+function generateEmbed(config, file) {
+  const targetPath = path.join(config.destDir, path.relative(config.src, file));
+  const document = parseDocument(file);
+  const sampleSections = document.sections.filter(
+    s => s.inBody && !s.isEmptyCodeSection()
+  );
+  sampleSections.forEach((section, index) => {
+    highlight(section.code).then(code => {
+      const tag = section.doc ? slug(section.doc.toLowerCase()) : index;
+      const context = {
+        sample: {
+          file: path.basename(targetPath),
+          preview: addFlag(targetPath, tag, 'preview'),
+          embed: addFlag(targetPath, tag, 'embed'),
+          template: addFlag(targetPath, tag, 'template'),
+          body: section.code,
+          code: code,
+        },
+        config: config,
+        document: document,
+      };
+      generate(context.sample.preview, templates.preview, context, /* minify */ true);
+      generate(context.sample.embed, templates.embed, context, /* minify */ true);
+      generateTemplate(context);
+    });
+  });
+}
+
+/**
+ * Syntax highlights a string.
+ */
+function highlight(code) {
+  return new Promise((resolve, reject) => {
+    pygmentize({ lang: 'html', format: 'html' }, code, function (err, result) {
+      if (err) {
+        console.log(err);
+        reject(err);
+      } else {
+        resolve(result.toString());
+      }
+    });
+  });
+}
+
+/**
+ * Renders the given template into a file.
+ */
+function generate(file, template, context, minifyResult) {
+  let string = template.render(context, {
+    'styles.css': templates.styles,
+    'embed.js': templates.embedJs
+  });
+  if (minifyResult) {
+    string = minify(string, {
+      caseSensitive: true,
+      collapseWhitespace: true,
+      html5: true,
+      minifyCSS: true,
+      minifyJS: true,
+      removeComments: true,
+      removeAttributeQuotes: true
+    });
+  }
+  writeFile(path.join(context.config.destRoot, file), string);
+}
+
+/**
+ * Appends a list of flags separated by a '.' to a filename.
+ */
+function addFlag() {
+  const filename = arguments[0];
+  const postfix = [].slice.call(arguments, 1).join('.');
+  return filename.replace('.html', '.' + postfix + '.html');
+}
+
+/**
+ * Parses an ABE document from a file.
+ */
+function parseDocument(file) {
+  const inputString = fs.readFileSync(file, 'utf-8');
+  return DocumentParser.parse(inputString);
+}
+
+/**
+ * Opens the embed html file in a browser to determine the initial height.
+ */
+function generateTemplate(context) {
+  let _page;
+  let _phantom;
+  phantom.create([], { logLevel: 'error' }).then(function (ph) {
```

Comment:
also, via @paulirish (thx!), this doc should be super helpful as a frame of context: https://medium.com/@dschnr/using-headless-chrome-as-an-automated-screenshot-tool-4b07dffba79a